crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -401,14 +624,45 @@ impl_param_set!();
 /// [`Exclusive`]: https://doc.rust-lang.org/nightly/std/sync/struct.Exclusive.html
 pub trait Resource: Send + Sync + 'static {}
 
+/// An error returned when a system fails to get a [`Res`].
+pub struct ResError<T: Resource> {
+    pub system_name: Cow<'static, str>,
+    _marker: PhantomData<T>,
+}
+
+impl<T: Resource> std::fmt::Debug for ResError<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ResError<{}> {{ system_name: {} }}",
+            std::any::type_name::<T>(),
+            self.system_name
+        )
```

Comment:
```suggestion
        f.debug_struct(&format!("ResError<{}>", std::any::type_name::<T>()))
            .field("system_name', &self.system_name)
            .finish()
```

Using the built-in formatter API will make this automatically work with different format configurations without any extra work on your part.